tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <counter.h>
+#include <ztest.h>
+#include <kernel.h>
+
+static volatile u32_t top_cnt;
+static volatile u32_t alarm_cnt;
+
+static void top_handler(struct device *dev, void *user_data);
+
+void *exp_user_data = (void *)199;
+
+#ifdef CONFIG_COUNTER_MCUX_RTC
```

Comment:
Why does MCUX require a special case, this feels like something is lacking in the API.